CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/TemplateUtilities.java

```diff
@@ -0,0 +1,167 @@
+import au.gov.asd.tac.constellation.plugins.importexport.translator.AttributeTranslator;
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameters;
+import au.gov.asd.tac.constellation.utilities.gui.NotifyDisplayer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+import javafx.scene.control.Alert;
+
+public class TemplateUtilities {
+    private static final String SOURCE = "source";
+    private static final String SCHEMA_INIT = "schema_init";
+    private static final String SHOW_ALL_SCHEMA_ATTRIBUTES = "show_all_schema_attributes";
+    private static final String DESTINATION = "destination";
+    private static final String DEFINITIONS = "definitions";
+    private static final String FILTER = "filter";
+    private static final String SCRIPT = "script";
+    private static final String COLUMNS = "columns";
+    private static final String FIRST_ROW = "first_row";
+    private static final String ATTRIBUTES = "attributes";
+    private static final String COLUMN_LABEL = "column_label";
+    private static final String ATTRIBUTE_LABEL = "attribute_label";
+    private static final String ATTRIBUTE_TYPE = "attribute_type";
+    private static final String ATTRIBUTE_DESCRIPTION = "attribute_description";
+    private static final String TRANSLATOR = "translator";
+    private static final String TRANSLATOR_ARGS = "translator_args";
+    private static final String DEFAULT_VALUE = "default_value";
+    private static final String LOAD_TEMPLATE = "Load Template";
+
+    private TemplateUtilities() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    /**
+     * This method is used to retrieve the essential fields from a template. If
+     * any is missing, a warning is displayed and NoSuchElementException is
+     * thrown.
+     */
+    public static JsonNode getRequiredFieldFromTemplate(final JsonNode jsonNode, final String fieldName) {
+        if (jsonNode.has(fieldName)) {
+            return jsonNode.get(fieldName);
+        } else {
+            //To handle npe when json parsor accessing missing required fields in the template
+            final String message = String.format("Missing field '%s' in the template. Add the missing field"
+                    + " manually or use a new template.", fieldName);
+            NotifyDisplayer.displayAlert(LOAD_TEMPLATE, "Template Error", message, Alert.AlertType.ERROR);
+            throw new NoSuchElementException(message);
+        }
+    }
+
+    public static List<ImportDefinition> getImportDefinitions(final ImportController importController, final JsonNode root, final String templName) {
+        final List<ImportDefinition> definitions = new ArrayList<>();
+        int tabCount = 0;
+        final JsonNode source = TemplateUtilities.getRequiredFieldFromTemplate(root, SOURCE);
```

Comment:
you should be able to call the function without the explicit call of `TemplateUtilities` since you are still operating in that class. Similarly for 2 of the next 3 lines after this one

Reply:
Oh initially added in ImportDelimitedIO.java. A copy paste issue! Fixed now.